lib/active_model/serializer.rb

```diff
@@ -195,7 +195,10 @@ def self.serializer_lookup_chain_for(klass)
       resource_namespace = klass.name.deconstantize
       serializer_class_name = "#{resource_class_name}Serializer"
 
-      chain.push("#{name}::#{serializer_class_name}") if self != ActiveModel::Serializer
+      if self != ActiveModel::Serializer
+        chain.push("#{name}::#{serializer_class_name}")
+        chain.push(*name.deconstantize.split('::').each_with_object([]) { |element, array| array.push((array.last ? array.last + '::' : '') + element) }.reverse.map { |k| k + "::#{serializer_class_name}" })
+      end
```

Comment:
@beauby another reason why search should be a detect rather than functional style without a stream to take from :)  as it is, it's making more calculations than it needs to.